### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by SukanyaSen530**
Components -
- Modal
- InputField
- Auth Modal

Pages -
- Minor style changes in home page

Features -
- Added signin and signup feature
- Prevented & protected routes

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -2,3 +2,6 @@
   text-align: center;
 }
 
+.Toastify__toast-container{
+  font-size: 1.5rem;
+}
```
Comment (JowelTisso) [inline src/App.css:7]: Is this BEM naming convention? Can you recheck your syntax.

Comment (SukanyaSen530) [reply src/App.css:7]: This class is present in the toastify component. Not my custom class.

**Review by sharathnair9999 — APPROVED** [review]

**Review by JowelTisso — APPROVED** [review]
* Clean code
* Rest looks good

File: src/App.css (hunk 1 above)
Comment (JowelTisso) [reply src/App.css:7]: ok

**Pull request merged**